hazelcast/src/main/java/com/hazelcast/util/OperatingSystemMXBeanSupport.java

```diff
@@ -38,18 +37,15 @@ private OperatingSystemMXBeanSupport() {
      * Reads a long attribute from OperatingSystemMXBean.
      *
      * @param attributeName name of the attribute
-     * @param defaultValue default value if the attribute value is null
+     * @param defaultValue  default value if the attribute value is null
      * @return value of the attribute
      */
     public static long readLongAttribute(String attributeName, long defaultValue) {
         try {
             String methodName = "get" + attributeName;
             OperatingSystemMXBean systemMXBean = OPERATING_SYSTEM_MX_BEAN;
             Method method = systemMXBean.getClass().getMethod(methodName);
-            // the method is public in Java 9
```

Comment:
Shouldn't private methods be accessed using `getDeclaredMethod` in the first place?

Reply:
yes and no :) The strange thing in this case is that the method is public... but it isn't.. So if you ask if it is public it will so no, but if you ask to print its access modifiers, it will say it is public + native.

So there is some very weird behavior going on. The method we are looking for 'is' public.